browser/js/global.js

```diff
@@ -675,6 +675,68 @@ window.app = {
 		}, false);
 	}
 
+	// indirect socket to wrap the asyncness around fetching the routetoken from indirection url endpoint
+	global.IndirectSocket = function(uri) {
+		var that = this;
+		this.uri = uri;
+		this.binaryType = '';
+		this.unloading = false;
+		this.readyState = 0; // connecting
+		this.innerSocket = undefined;
+
+		this.onclose = function() {};
+		this.onerror = function () {};
+		this.onmessage = function () {};
+		this.onopen = function () {};
+
+		this.close = function() {
+			this.readyState = 3;
+			this.innerSocket.onerror = function () {};
+			this.innerSocket.onclose = function () {};
+			this.innerSocket.onmessage = function () {};
+			this.innerSocket.close();
+		};
+
+		this.send = function(msg) {
+			this.innerSocket.send(msg);
+		};
+
+		this.setUnloading = function() {
+			this.unloading = true;
+		};
+
+		var http = new XMLHttpRequest();
+		http.open('GET', global.indirectionUrl, true);
+		http.responseType = 'json';
+		http.addEventListener('load', function() {
+			if (this.status === 200) {
+				global.routeToken = http.response.route_token;
+				//add routeToken in uri
+				var startIndex = that.uri.indexOf('&compat');
```

Comment:
I would really prefer to get the full URL back from the server including the routing token - rather than adding bits into the URI like this. ie. lets do this on the server side. Can we do that ? =)